src/web/components/MostViewed/MostViewedRight/MostViewedRight.tsx

```diff
@@ -46,26 +35,22 @@ export const MostViewedRight = ({ pillar, limitItems = 5 }: Props) => {
 	}
 
 	if (data) {
+		const trails: TrailType[] = data.trails
+			.map(decideTrail)
+			.slice(0, limitItems);
 		// Look I don't know why data-component is geo-most-popular either, but it is, ok? Ok.
 		return (
 			<div className={wrapperStyles} data-component="geo-most-popular">
 				<GuardianLines count={4} pillar={pillar} />
 				<h3 className={headingStyles}>Most viewed</h3>
 				<ul data-link-name="Right hand most popular geo GB">
-					{(data.trails || [])
-						.slice(0, limitItems)
-						.map(
-							(
-								trail: CAPITrailType,
-								mostViewedItemIndex: number,
-							) => (
-								<MostViewedRightItem
-									key={trail.url}
-									trail={transformTrail(trail)}
-									mostViewedItemIndex={mostViewedItemIndex}
-								/>
-							),
-						)}
+					{(trails || []).map((trail, index) => (
```

Comment:
if `data.trails` is not an array it would fail at line 38, maybe do the cheak for its existance before? or remove ` || []` if you know its defined

Reply:
Yeah, I was tempted to remove this. I suspect the reason it was put there to start with was typescript related